### DISCUSSION TITLE
How to train the model on own dataset?

### DISCUSSION THREAD

**Discussion by ghost** [mindee/doctr#581] [Q&A] [answered]
I want to train the model on my own dataset How can I train the model?

**Comment by charlesmindee** [comment, marked as answer]
Hi again @hammadyounas2008,

In the `references` section of the repo you will find training scripts for the detection/recognition and classification tasks (for both frameworks, tensorflow and pytorch). In the readme of those sections ([there for instance](https://github.com/mindee/doctr/tree/main/references/detection)) we explain how to use those scripts (data format, options, ...).

I hope this answer your question !

**Reply by ghost** [reply]
I have tried it but it not working,  How to train the FUNSD  Dataset?

**Reply by ghost** [reply]
I have the label the dataset using VOTT
My JSON File is as shown below
```
 {
    "asset": {
        "format": "jpeg",
        "id": "72fbf4e3b8443331e5ea105cf2e870d1",
        "name": "passbook_68_0.jpeg",
        "path": "file:C:/Freelancing/Newwork/New%20folder%20(2)/passbook_68_0.jpeg",
        "size": {
            "width": 500,
            "height": 235
        },
        "state": 2,
        "type": 1
    },
    "regions": [
        {
            "id": "Lx8zjHIeW",
            "type": "POLYGON",
            "tags": [
                "Word"
            ],
            "boundingBox": {
                "height": 16.70616113744076,
                "width": 71.90635451505017,
                "left": 377.3690078037904,
                "top": 19.377406694312796
            },
            "points": [
                {
                    "x": 377.9264214046823,
                    "y": 22.71863892180095
                },
                {
                    "x": 377.9264214046823,
                    "y": 32.742335604265406
                },
                {
                    "x": 449.27536231884056,
                    "y": 36.083567831753555
                },
                {
                    "x": 449.27536231884056,
                    "y": 23.275510959715643
                },
                {
                    "x": 377.3690078037904,
                    "y": 19.377406694312796
                }
            ]
        },
        {
            "id": "i48JKyGHc",
            "type": "POLYGON",
            "tags": [
                "Word"
            ],
            "boundingBox": {
                "height": 11.694312796208532,
                "width": 90.85841694537346,
                "left": 303.2329988851728,
                "top": 62.81342565165877
            },
            "points": [
                {
                    "x": 304.9052396878484,
                    "y": 63.927169727488156
                },
                {
                    "x": 304.9052396878484,
                    "y": 73.39399437203792
                },
                {
                    "x": 394.0914158305463,
                    "y": 74.5077384478673
                },
                {
                    "x": 394.0914158305463,
                    "y": 64.48404176540285
                },
                {
                    "x": 303.2329988851728,
                    "y": 62.81342565165877
                }
            ]
        },
```

**Reply by fg-mindee** [reply]
Hi @hammadyounas2008 !

About FUNSD, it's not meant to be used for training in docTR (we consider it to be too small to get something robust) so it's expected for now!

About your own dataset label format, those discussions are starting to be all duplicate of each other, please pay attention :pray: 
We already provided you with the proper README for data format here https://github.com/mindee/doctr/discussions/585, and as mentioned, this is the target format, making the conversion is up to you :+1: 

Cheers!

**Reply by ghost** [reply]
@fg-mindee  so you don't use FUNSD dataset for training?

**Reply by fg-mindee** [reply]
We don't since we use it for testing purposes (cf. benchmark on public datasets https://mindee.github.io/doctr/latest/using_models.html#id3)
and it's rather small to get the generic & robust OCR that we want to build with docTR :)

**Reply by ghost** [reply]
Means we cannot train this repo?

**Reply by fg-mindee** [reply]
?
If you've read the READMEs of the training scripts, you should know that you can train models with those scripts. For character classification, we included a data generator (working on-the-fly) so you don't need any data. For the rest, as mentioned in those instructions and in https://github.com/mindee/doctr/discussions/585, you need to have your own dataset and labels formatted to the specified format.

If you wish to use public datasets supported by docTR for training, it's your choice, you might have to adapt the label format to meet the training expectations
